edge/pkg/edged/edged.go

```diff
@@ -473,6 +540,42 @@ func newEdged() (*edged, error) {
 				conf.remoteImageEndpoint = conf.remoteRuntimeEndpoint
 			}
 		}
+		//create and start the docker shim running as a grpc server
+		if conf.remoteRuntimeEndpoint == DockerShimEndpoint || conf.remoteRuntimeEndpoint == DockerShimEndpointDeprecated {
```

Comment:
why we need to support DockerShimEndpointDeprecated ?

Reply:
Dockershim is still supported, and there will be a waring log

Reply:
yes，

> there will be no DNS available in most of the edge cases, can we handle that with this code? I saw the CNI initialization in the code, does that impact our current non-CNI situation?

We can configure cluster dns to be None and let ‘’clusterDNS‘’ take effect. And I think the cni initlization of the code is in order to fill the parameters and greatly make sure that there is no null pointer error for some method call